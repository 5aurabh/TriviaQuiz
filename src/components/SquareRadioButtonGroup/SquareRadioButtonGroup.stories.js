import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import SquareRadioButtonGroup from './index.js';

storiesOf('SquareRadioButtonGroup', module)
  .add('default', () => (
    <SquareRadioButtonGroup
      labels={['Apple', 'Ball', 'Cat']}
      values={['1', '2', '3']}
    />
  ))
  .add('with default checked', () => (
    <SquareRadioButtonGroup
      labels={['Choice 1', 'choice 2']}
      values={['1', '2']}
      defaultChecked="1"
    />
  ))
  .add('with custom handler checked', () => (
    <SquareRadioButtonGroup
      labels={['OSX', 'Windows']}
      values={['1', '2']}
      defaultChecked="1"
      onChange={action('change handler')}
    />
  ));
