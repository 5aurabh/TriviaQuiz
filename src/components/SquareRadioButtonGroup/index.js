// @flow
import React from 'react';
import styles from './index.css';
import { SquareRadioButton } from '../index';
type Props = {
  values: Array<string>,
  labels: Array<string>,
  defaultChecked: ?string,
  name: ?string,
  onChange: () => *,
};
const SquareRadioButtonGroup = (props: Props) => {
  return (
    <div className={styles.SquareRadioButtonGroup}>
      {props.values.map((value, index) => (
        <SquareRadioButton
          value={value}
          id={index}
          name={props.name}
          label={props.labels[index]}
          checked={props.defaultChecked === value}
          onChange={props.onChange}
        />
      ))}
    </div>
  );
};

/* Add default props */
SquareRadioButtonGroup.defaultProps = {
  values: [],
  labels: [],
  defaultChecked: null,
  name: 'standard-name',
  onChange: () => {},
};

export default SquareRadioButtonGroup;
