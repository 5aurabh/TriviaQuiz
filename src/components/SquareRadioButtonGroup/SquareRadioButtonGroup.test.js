import React from 'react';
import renderer from 'react-test-renderer';
import SquareRadioButtonGroup from './index.js';
import { SquareRadioButton } from '../index';
import { shallow, mount } from 'enzyme';

const commonProps = {
  values: ['1', '2', '3'],
  labels: ['yes', 'no', 'maybe'],
};

it('Renders SquareRadioButtonGroup successfully', () => {
  const component = renderer.create(
    <SquareRadioButtonGroup {...commonProps} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correct number of SquareRadio buttons', () => {
  const props = Object.assign(commonProps, { defaultChecked: '2' });
  const wrapper = shallow(<SquareRadioButtonGroup {...props} />);
  expect(wrapper.find(SquareRadioButton)).toHaveLength(3);
});

it('Renders radio buttons with checked state based on defaultChecked prop', () => {
  const props = Object.assign(commonProps, { defaultChecked: '2' });
  const wrapper = mount(<SquareRadioButtonGroup {...props} />);
  const radioBoxes = wrapper.find('input[type="radio"]');
  const checkedRadioButton = radioBoxes.at(1);
  expect(checkedRadioButton.props()['value']).toEqual('2');
});
