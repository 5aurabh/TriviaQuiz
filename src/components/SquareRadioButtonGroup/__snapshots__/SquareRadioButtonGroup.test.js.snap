// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders SquareRadioButtonGroup successfully 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <input
      className={undefined}
      defaultChecked={false}
      id={0}
      name="standard-name"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <label
      for={0}
    >
      yes
    </label>
  </div>
  <div
    className={undefined}
  >
    <input
      className={undefined}
      defaultChecked={false}
      id={1}
      name="standard-name"
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <label
      for={1}
    >
      no
    </label>
  </div>
  <div
    className={undefined}
  >
    <input
      className={undefined}
      defaultChecked={false}
      id={2}
      name="standard-name"
      onChange={[Function]}
      type="radio"
      value="3"
    />
    <label
      for={2}
    >
      maybe
    </label>
  </div>
</div>
`;
